index.js

```diff
@@ -100,185 +113,209 @@ const VerifyOptions = t.interface({
+      algorithm: algorithm
+    });
 
   /**
    * Issues a new access token using a refresh token and an old token.
    * There is no need to verify the old token provided because this method uses the stored one.
    * @param {String} refreshToken
    * @param {String} oldToken
-   * @param secret
-   * @param {UserSignOptions} [signOptions] Options passed to jwt.sign
+   * @param {Secret} secret
+   * @param {SignOptions} [signOptions] Options passed to jwt.sign
    * @returns {Promise<Tokens>}
-   * @throws {RefreshTokenExpiredOrNotFound} RefreshTokenExpiredOrNotFound
-   * @throws {InvalidAccessToken} InvalidAccessToken
+   * @throws {RefreshTokenExpiredOrNotFound} refreshTokenExpiredOrNotFound
+   * @throws {InvalidAccessToken} invalidAccessToken
    */
-  async refresh(refreshToken, oldToken, secret, signOptions) {
+  const refresh = async (refreshToken, oldToken, secret) => {
     RefreshToken(refreshToken);
     Token(oldToken);
-    const untrustedPayload = Payload(this._jwt.decode(oldToken));
-    const trustedToken =
-      await this._store.getAccessToken(untrustedPayload.pld.userId, refreshToken);
+
+    const untrustedPayload = Payload(jwt.decode(oldToken));
+    const trustedToken = await store.getAccessToken(untrustedPayload.uid, refreshToken);
     // Remove the refresh token even if the following operations were not successful.
     // RefreshTokens are one time use only
-    if(!await this._store.remove(untrustedPayload.pld.userId, refreshToken)) {
-      throw RefreshTokenExpiredOrNotFound;
+    if(!await store.remove(untrustedPayload.uid, refreshToken)) {
+      throw refreshTokenExpiredOrNotFound;
     }
     // RefreshTokens works with only one AccessToken
-    if (trustedToken !== oldToken) {throw InvalidAccessToken;}
+    if (trustedToken !== oldToken) {throw invalidAccessToken;}
 
     // Token is safe since it is stored by us
-    const {pld: payload, rme: rememberMe, ...jwtOptions} =
-      this._jwt.decode(trustedToken);
+    // eslint-disable-next-line no-unused-vars
+    const {exp, iat, uid, pld: payload, rme: prolong, ...jwtOptions} = jwt.decode(trustedToken);
 
     // Finally, sign new tokens for the user
-    return this.sign(
-      payload,
+    return sign(
+      uid,
       Secret(secret),
-      rememberMe,
-      // Ignoring exp
-      UserSignOptions({...omit(['exp', 'iat'], jwtOptions), ...signOptions})
+      payload,
+      prolong,
+      jwtOptions
     );
-  }
+  };
 
   /**
    * Invalidates refresh token
-   * @param {String|Number} userId
    * @param {String} refreshToken
    * @returns {Promise<Number>} 1 if successful, 0 otherwise.
    */
-  invalidateRefreshToken(userId, refreshToken) {
-    return this._store.remove(UserId(userId), RefreshToken(refreshToken));
-  }
+  const invalidateRefreshToken = refreshToken => {
+    RefreshToken(refreshToken);
+    const userId = getUserIdFromRefreshToken(refreshToken);
+    return store.remove(userId, refreshToken);
+  };
```

Comment:
This function can be simplified to: `refreshToken => store.remove(refreshToken)` given the suggestion above.